src/api/auth/login.ts

```diff
@@ -1,15 +1,88 @@
 import { Express, Response, Request, NextFunction } from 'express'
 import passport from 'passport'
-import { User } from '@openforis/arena-core'
+import { Authorizer, ServiceRegistry, ServiceType, SurveyService, User, UserService } from '@openforis/arena-core'
 
 import { ExpressInitializer } from '../../server'
 import { ApiEndpoint } from '../endpoint'
+import { Logger } from '../../log'
+import { AuthGroupRepository } from '../../repository'
+
+const logger = new Logger('AuthAPI')
+
+export const updateUserPrefs = async (userToUpdate: User): Promise<User> => {
+  const service = ServiceRegistry.getInstance().getService(ServiceType.user) as UserService
+  const user = await service.updateUserPrefs({ userToUpdate })
+  return {
+    ...user,
+    authGroups: await AuthGroupRepository.getMany({ userUuid: user.uuid }),
+  }
+}
+
+export const deletePrefSurvey = (user: User): User => {
+  const surveyId = user.prefs?.surveys?.current
+  if (!surveyId) return user
+  const current = { current: -1 }
+  const surveys = {
+    ...user.prefs?.surveys,
+    [-1]: { cycle: 0 },
+  }
+
+  const _user: User = {
+    ...user,
+    prefs: {
+      ...user.prefs,
+      surveys: {
+        ...surveys,
+        ...current,
+      },
+    },
+  }
+  delete _user.prefs?.surveys?.[surveyId]
+  return _user
+}
+
+const sendResponse = (res: Response, user: User) => res.json({ user })
+
+const sendUserSurvey = async (options: { res: Response; user: User }) => {
+  const { res, user } = options
+  const surveyId = user.prefs?.surveys?.current
+  try {
+    const service = ServiceRegistry.getInstance().getService(ServiceType.survey) as SurveyService
+    let survey = null
+    if (surveyId) {
+      survey = await service.get({ surveyId, draft: false, validate: false })
+    }
+
+    if (survey && surveyId && Authorizer.canEditSurvey(user, survey)) {
```

Comment:
this can just be 
```
if (survey && Authorizer.canEditSurvey(user, survey)) {
```

Reply:
surveyId is checked here because next line uses surveyId where it can't be null